### PR TITLE
Added doxygen code documentation

### PR DESCRIPTION
**Pull request by brettsalyer**
Added some doxygen code documentation for the  `input.h` file

### CODE REVIEW

File: input/input.h [hunk 1]
```diff
@@ -11,7 +11,26 @@
 
 #include "defines.h"
 
-
+/**
+ * Waits for user input and returns the input
+ * as an ASCII character
+ */
```
Comment (Kwask) [inline input/input.h:19]: Needs an `@return` to say what the return value is

File: input/input.h [hunk 2]
```diff
@@ -11,7 +11,26 @@
...
 char get_keystroke_ascii();
+
+/**
+ * Waits for the user input and returns the input
+ * as a scancode. This is used for capturing keystrokes 
+ * that are not representable with an ASCII character.
+ * 
+ */
 char get_keystroke_scancode();
```
Comment (Kwask) [inline input/input.h:29]: Same here, needs `@return`

File: input/input.h [hunk 3]
```diff
@@ -11,7 +11,26 @@
...
+
+/**
+ * Compares the input the user has entered with
+ * some other character, whether that be ASCII
+ * or scan code. 
+ * 
+ * @param input the input the user as entered
+ * @param comparing the scan code or ASCII character that you wish to check for
+ */
```
Comment (Kwask) [inline input/input.h:40]: and also same for this one, `@return`

**Review by Kwask — CHANGES_REQUESTED** [review]
Needs an `@return` for the three functions that have a return value describing what the return value represents.

File: input/input.h (hunk 3 above)
Comment (brettsalyer) [reply input/input.h:40]: Yay! The first rejected code!

File: input/input.h (hunk 1 above)
Comment (brettsalyer) [reply input/input.h:19]: Fixed

File: input/input.h (hunk 3 above)
Comment (brettsalyer) [reply input/input.h:40]: Fixed

File: input/input.h (hunk 2 above)
Comment (brettsalyer) [reply input/input.h:29]: Fixed

**Review by Kwask — APPROVED** [review]
Everything looks good to me now :+1:

**Pull request merged**